<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />



<title>FPSOmics: Ferroptosis Score and associated multi-Omics</title>

<script src="data:application/javascript;base64,Ly8gUGFuZG9jIDIuOSBhZGRzIGF0dHJpYnV0ZXMgb24gYm90aCBoZWFkZXIgYW5kIGRpdi4gV2UgcmVtb3ZlIHRoZSBmb3JtZXIgKHRvCi8vIGJlIGNvbXBhdGlibGUgd2l0aCB0aGUgYmVoYXZpb3Igb2YgUGFuZG9jIDwgMi44KS4KZG9jdW1lbnQuYWRkRXZlbnRMaXN0ZW5lcignRE9NQ29udGVudExvYWRlZCcsIGZ1bmN0aW9uKGUpIHsKICB2YXIgaHMgPSBkb2N1bWVudC5xdWVyeVNlbGVjdG9yQWxsKCJkaXYuc2VjdGlvbltjbGFzcyo9J2xldmVsJ10gPiA6Zmlyc3QtY2hpbGQiKTsKICB2YXIgaSwgaCwgYTsKICBmb3IgKGkgPSAwOyBpIDwgaHMubGVuZ3RoOyBpKyspIHsKICAgIGggPSBoc1tpXTsKICAgIGlmICghL15oWzEtNl0kL2kudGVzdChoLnRhZ05hbWUpKSBjb250aW51ZTsgIC8vIGl0IHNob3VsZCBiZSBhIGhlYWRlciBoMS1oNgogICAgYSA9IGguYXR0cmlidXRlczsKICAgIHdoaWxlIChhLmxlbmd0aCA+IDApIGgucmVtb3ZlQXR0cmlidXRlKGFbMF0ubmFtZSk7CiAgfQp9KTsK"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="data:text/css,body%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Amargin%3A%201em%20auto%3B%0Amax%2Dwidth%3A%20700px%3B%0Aoverflow%3A%20visible%3B%0Apadding%2Dleft%3A%202em%3B%0Apadding%2Dright%3A%202em%3B%0Afont%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Afont%2Dsize%3A%2014px%3B%0Aline%2Dheight%3A%201%2E35%3B%0A%7D%0A%23TOC%20%7B%0Aclear%3A%20both%3B%0Amargin%3A%200%200%2010px%2010px%3B%0Apadding%3A%204px%3B%0Awidth%3A%20400px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Aborder%2Dradius%3A%205px%3B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Afont%2Dsize%3A%2013px%3B%0Aline%2Dheight%3A%201%2E3%3B%0A%7D%0A%23TOC%20%2Etoctitle%20%7B%0Afont%2Dweight%3A%20bold%3B%0Afont%2Dsize%3A%2015px%3B%0Amargin%2Dleft%3A%205px%3B%0A%7D%0A%23TOC%20ul%20%7B%0Apadding%2Dleft%3A%2040px%3B%0Amargin%2Dleft%3A%20%2D1%2E5em%3B%0Amargin%2Dtop%3A%205px%3B%0Amargin%2Dbottom%3A%205px%3B%0A%7D%0A%23TOC%20ul%20ul%20%7B%0Amargin%2Dleft%3A%20%2D2em%3B%0A%7D%0A%23TOC%20li%20%7B%0Aline%2Dheight%3A%2016px%3B%0A%7D%0Atable%20%7B%0Amargin%3A%201em%20auto%3B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dcolor%3A%20%23DDDDDD%3B%0Aborder%2Dstyle%3A%20outset%3B%0Aborder%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0Aborder%2Dwidth%3A%202px%3B%0Apadding%3A%205px%3B%0Aborder%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dstyle%3A%20inset%3B%0Aline%2Dheight%3A%2018px%3B%0Apadding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0Aborder%2Dleft%2Dstyle%3A%20none%3B%0Aborder%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Ap%20%7B%0Amargin%3A%200%2E5em%200%3B%0A%7D%0Ablockquote%20%7B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Apadding%3A%200%2E25em%200%2E75em%3B%0A%7D%0Ahr%20%7B%0Aborder%2Dstyle%3A%20solid%3B%0Aborder%3A%20none%3B%0Aborder%2Dtop%3A%201px%20solid%20%23777%3B%0Amargin%3A%2028px%200%3B%0A%7D%0Adl%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Adl%20dd%20%7B%0Amargin%2Dbottom%3A%2013px%3B%0Amargin%2Dleft%3A%2013px%3B%0A%7D%0Adl%20dt%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Aul%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Aul%20li%20%7B%0Alist%2Dstyle%3A%20circle%20outside%3B%0A%7D%0Aul%20ul%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Apre%2C%20code%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0Acolor%3A%20%23333%3B%0Awhite%2Dspace%3A%20pre%2Dwrap%3B%20%0A%7D%0Apre%20%7B%0Aborder%2Dradius%3A%203px%3B%0Amargin%3A%205px%200px%2010px%200px%3B%0Apadding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0Afont%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0Apadding%3A%202px%200px%3B%0A%7D%0Adiv%2Efigure%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0Abackground%2Dcolor%3A%20%23FFFFFF%3B%0Apadding%3A%202px%3B%0Aborder%3A%201px%20solid%20%23DDDDDD%3B%0Aborder%2Dradius%3A%203px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Amargin%3A%200%205px%3B%0A%7D%0Ah1%20%7B%0Amargin%2Dtop%3A%200%3B%0Afont%2Dsize%3A%2035px%3B%0Aline%2Dheight%3A%2040px%3B%0A%7D%0Ah2%20%7B%0Aborder%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Apadding%2Dbottom%3A%202px%3B%0Afont%2Dsize%3A%20145%25%3B%0A%7D%0Ah3%20%7B%0Aborder%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Afont%2Dsize%3A%20120%25%3B%0A%7D%0Ah4%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0Amargin%2Dleft%3A%208px%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Ah5%2C%20h6%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23ccc%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%230033dd%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%20%7B%0Acolor%3A%20%236666ff%3B%20%7D%0Aa%3Avisited%20%7B%0Acolor%3A%20%23800080%3B%20%7D%0Aa%3Avisited%3Ahover%20%7B%0Acolor%3A%20%23BB00BB%3B%20%7D%0Aa%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0Aa%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%0A" type="text/css" />




</head>

<body>




<h1 class="title toc-ignore">FPSOmics: Ferroptosis Score and associated multi-Omics</h1>


<div id="TOC">
<ul>
<li><a href="#introduction">1. Introduction</a></li>
<li><a href="#installation-and-requirement">2. Installation and requirement</a></li>
<li><a href="#functional-modules">3. Functional modules</a>
<ul>
<li><a href="#calculate-ferroptosis-score">3.1 Calculate ferroptosis score</a></li>
<li><a href="#evaluate-prognostic-significance">3.2 Evaluate prognostic significance</a></li>
<li><a href="#multidimensional-ferroptosis-associated-molecular-signatures">3.3 Multidimensional ferroptosis-associated molecular signatures</a></li>
<li><a href="#function-enrichment">3.4 Function enrichment</a></li>
</ul></li>
<li><a href="#summary">4. Summary</a></li>
<li><a href="#session-information">5. Session information</a></li>
<li><a href="#citing-fpsomics">6. Citing FPSOmics</a></li>
<li><a href="#references">7. REFERENCES</a></li>
</ul>
</div>

<div id="introduction" class="section level2">
<h2>1. Introduction</h2>
<p>FPSOmics is an R package to calculate ferroptosis score (FPS) and multidimensional ferroptosis-associated molecular signatures. FPSOmics designed a novel score model based on ferroptosis-related genes(FRGs)<span class="citation"><sup>1</sup></span> using single sample gene set enrichment analysis (ssGSEA)<span class="citation"><sup>2</sup></span>. This package also include a propensity score matching (PSM) algorithm<span class="citation"><sup>3</sup></span>, which appropriately control the effects of clinical confounding factors, to calculate ferroptosis-associated molecular signatures. The full FPSOmics documentation is available in the package. To reach the user’s Guide, install the FPSOmics package and load it into an R session by library (FPSOmics). And they can get help by help (FPS), help (FPSsurvival), help (FPSmultiOmicsSig) and help (FPSmultiOmicsSigEnrich) to see documentation of each function. Also description of each function is including in docs/FPSOmics_0.1.0.pdf.</p>
</div>
<div id="installation-and-requirement" class="section level2">
<h2>2. Installation and requirement</h2>
<p>Before installing FPSOmics, please properly install all dependencies</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co">#library(magrittr)</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="co">#library(survival)</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="co">#library(survminer)</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="co">#library(ggplot2)</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="co">#library(grid)</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="co">#library(gridExtra)</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="co">#library(gtable)</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="co">#library(org.Hs.eg.db)</span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="co">#library(KEGG.db)</span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="co">#library(fgsea)</span></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="co">#library(enrichplot)</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="co">#library(clusterProfiler)</span></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="co">#library(magrittr)</span></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="co">#library(dummies)</span></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="co">#library(formula.tools)</span></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="co">#library(doMC)</span></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="co">#library(foreach)</span></span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a><span class="co">#library(gdata)</span></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a><span class="co">#library(gtools)</span></span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a><span class="co">#library(gmodels)</span></span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a><span class="co">#library(MASS)</span></span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a><span class="co">#library(gplots)</span></span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a><span class="co">#library(formula.tools)</span></span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a><span class="co">#library(Hmisc) </span></span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a><span class="co">#library(Matrix)</span></span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a><span class="co">#library(dplyr)</span></span></code></pre></div>
<p>Installation of FPSOmics You can install FPSOmics.tar.gz from local path and directly from github.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co">#devtools::install_github(&#39;Yelab2020/FPSOmics&#39;)</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="co">#install.packages(&#39;FPSOmics.tar.gz&quot;, repos = NULL, type = &quot;source&quot;)</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="co">#library(FPSOmics)</span></span></code></pre></div>
</div>
<div id="functional-modules" class="section level2">
<h2>3. Functional modules</h2>
<div id="calculate-ferroptosis-score" class="section level3">
<h3>3.1 Calculate ferroptosis score</h3>
<p>We firstly filtered out to 32 core ferroptosis-related genes (FRGs) from genelist in FerrDb<span class="citation"><sup>1</sup></span>.Then, we calculate enrichment score (ES) of ferroptosis-promote genes (pro-FRGs) and ferroptosis-inhibit genes(anti-FRGs) using single sample gene set enrichment analysis (ssGSEA)<span class="citation"><sup>2</sup></span>, then ferroptosis score (FPS) was defined by the ES of pro-FRGs minus anti-FRGs.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">&#39;FPS&#39;</span>)</span></code></pre></div>
<pre><code>## [1] &quot;FPS&quot;</code></pre>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co">#data(m1_input_mRNA,package=&#39;FPSOmics&#39;)</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="co">#FPS_score=FPSOmics::FPS(m1_input_mRNA)</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="co">#head(FPS_score)</span></span></code></pre></div>
</div>
<div id="evaluate-prognostic-significance" class="section level3">
<h3>3.2 Evaluate prognostic significance</h3>
<p>We firstly calculate enrichment score (ES) of ferroptosis-promote genes (pro-FRGs) and ferroptosis-inhibit genes (anti-FRGs) using single sample gene set enrichment analysis (ssGSEA), then ferroptosis score (FPS) was defined by the ES of pro-FRGs minus anti-FRGs. Then Kaplan-Meier survival analyses between high-FPS and low-FPS were also assessed using log rank test.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">&#39;FPS-survival&#39;</span>)</span></code></pre></div>
<pre><code>## [1] &quot;FPS-survival&quot;</code></pre>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co">#data(m1_input_mRNA,package=&#39;FPSOmics&#39;)#load mRNA matrix</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="co">#data(m2_input_survival_data,package=&#39;FPSOmics&#39;) #load survival time</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="co">#FPS_survival_result=FPSOmics::FPSsurvival(m1_input_mRNA,m2_input_survival_data)</span></span></code></pre></div>
</div>
<div id="multidimensional-ferroptosis-associated-molecular-signatures" class="section level3">
<h3>3.3 Multidimensional ferroptosis-associated molecular signatures</h3>
<p>A propensity score matching (PSM) algorithm<span class="citation"><sup>3</sup></span>, which appropriately control the effects of clinical confounding factors, was used to calculate ferroptosis-associated molecular signatures between high and low ferroptosis score group.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">&#39;FPS-multiOmics&#39;</span>)</span></code></pre></div>
<pre><code>## [1] &quot;FPS-multiOmics&quot;</code></pre>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co">#data(m3_input_Clinical,package=&#39;FPSOmics&#39;)</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="co">#data(m3_input_FPS_score,package=&#39;FPSOmics&#39;)</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="co">#data(m3_input_mRNA,package=&#39;FPSOmics&#39;)</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="co">#PSM_result=FPSOmics::FPSmultiOmicsSig(m3_input_Clinical,m3_input_FPS_score,m3_input_mRNA)</span></span></code></pre></div>
</div>
<div id="function-enrichment" class="section level3">
<h3>3.4 Function enrichment</h3>
<p>Study function of ferroptosis-associated molecular signatures and the differences of different ferroptosis status in cancer hallmarks, KEGG pathway by Gene Set Enrichment Analysis (GSEA)<span class="citation"><sup>4,5</sup></span> and GO, KEGG enrichment by clusterprofile<span class="citation"><sup>6</sup></span>.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">&#39;FPS-multiOmics-function&#39;</span>)</span></code></pre></div>
<pre><code>## [1] &quot;FPS-multiOmics-function&quot;</code></pre>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co">#data(m4_input_gs_kegg,package=&#39;FPSOmics&#39;)</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="co">#data(m4_input_PSM,package=&#39;FPSOmics&#39;)</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a><span class="co">#enrichment_result=FPSOmics::FPSmultiOmicsSigEnrich(m4_input_PSM,m4_input_gs_kegg,&#39;temp_folder&#39;)</span></span></code></pre></div>
</div>
</div>
<div id="summary" class="section level2">
<h2>4. Summary</h2>
<p>FPS provides four function modules which allows users to analysis of ferroptosis score and their-related multi-omics molecular signatures. There may still be gaps in the toolkit, but we’ll work on more features later. If you have any suggestions or questions, please feel free to contact us (<a href="mailto:yihe0902@163.com" class="email">yihe0902@163.com</a> or <a href="mailto:yudong123@sjtu.edu.cn" class="email">yudong123@sjtu.edu.cn</a>).</p>
</div>
<div id="session-information" class="section level2">
<h2>5. Session information</h2>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 3.6.0 (2019-04-26)
## Platform: x86_64-redhat-linux-gnu (64-bit)
## Running under: CentOS Linux 7 (Core)
## 
## Matrix products: default
## BLAS/LAPACK: /usr/lib64/R/lib/libRblas.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## loaded via a namespace (and not attached):
##  [1] rstudioapi_0.11   knitr_1.37        magrittr_2.0.1    usethis_1.6.3    
##  [5] devtools_2.3.2    pkgload_1.1.0     R6_2.4.1          rlang_0.4.11     
##  [9] fastmap_1.1.0     stringr_1.4.0     tools_3.6.0       pkgbuild_1.1.0   
## [13] xfun_0.29         sessioninfo_1.1.1 cli_3.0.1         jquerylib_0.1.4  
## [17] withr_2.3.0       htmltools_0.5.2   ellipsis_0.3.2    remotes_2.2.0    
## [21] yaml_2.2.1        digest_0.6.27     assertthat_0.2.1  rprojroot_2.0.2  
## [25] crayon_1.3.4      processx_3.5.2    callr_3.7.0       sass_0.4.0       
## [29] fs_1.5.0          ps_1.3.4          testthat_3.1.1    memoise_1.1.0    
## [33] glue_1.4.2        evaluate_0.14     rmarkdown_2.9     stringi_1.5.3    
## [37] bslib_0.3.1       compiler_3.6.0    desc_1.2.0        prettyunits_1.1.1
## [41] jsonlite_1.7.1</code></pre>
</div>
<div id="citing-fpsomics" class="section level2">
<h2>6. Citing FPSOmics</h2>
<p>If FPSOmics R package is used in your research, please cite: Multi-omics characterization and therapeutic liability of ferroptosis in melanoma</p>
</div>
<div id="references" class="section level2 unnumbered">
<h2 class="unnumbered">7. REFERENCES</h2>
<div id="refs" class="references csl-bib-body" line-spacing="2">
<div id="ref-Zhou2020" class="csl-entry">
<div class="csl-left-margin">1. </div><div class="csl-right-inline">Zhou, N. &amp; Bao, J. <span class="nocase">FerrDb: a manually curated resource for regulators and markers of ferroptosis and ferroptosis-disease associations</span>. <em>Database</em> <strong>2020</strong>, 21 (2020).</div>
</div>
<div id="ref-Hanzelmann2013" class="csl-entry">
<div class="csl-left-margin">2. </div><div class="csl-right-inline">Hänzelmann, S., Castelo, R. &amp; Guinney, J. <span class="nocase">GSVA: gene set variation analysis for microarray and RNA-Seq data</span>. <em>BMC Bioinformatics</em> <strong>14</strong>, 7 (2013).</div>
</div>
<div id="ref-Ye2019" class="csl-entry">
<div class="csl-left-margin">3. </div><div class="csl-right-inline">Ye, Y. <em>et al.</em> <span class="nocase">Characterization of hypoxia-associated molecular features to aid hypoxia-targeted therapy</span>. <em>Nature Metabolism</em> <strong>1</strong>, 431–444 (2019).</div>
</div>
<div id="ref-Sergushichev2016" class="csl-entry">
<div class="csl-left-margin">4. </div><div class="csl-right-inline">Sergushichev, A. A. <span class="nocase">An algorithm for fast preranked gene set enrichment analysis using cumulative statistic calculation</span>. <em>bioRxiv</em> 60012 (2016) doi:<a href="https://doi.org/10.1101/060012">10.1101/060012</a>.</div>
</div>
<div id="ref-Subramanian2005" class="csl-entry">
<div class="csl-left-margin">5. </div><div class="csl-right-inline">Subramanian, A. <em>et al.</em> <span class="nocase">Gene set enrichment analysis: A knowledge-based approach for interpreting genome-wide expression profiles</span>. <em>Proceedings of the National Academy of Sciences of the United States of America</em> <strong>102</strong>, 15545–15550 (2005).</div>
</div>
<div id="ref-Yu2012" class="csl-entry">
<div class="csl-left-margin">6. </div><div class="csl-right-inline">Yu, G., Wang, L.-G., Han, Y. &amp; He, Q.-Y. <span class="nocase">clusterProfiler: an R package for comparing biological themes among gene clusters.</span> <em>Omics : a journal of integrative biology</em> <strong>16</strong>, 284–287 (2012).</div>
</div>
</div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
